.pre-commit-config.yaml

```
@@ -26,7 +26,7 @@ repos:
     hooks:
       - id: pyupgrade
         exclude: ^fuzz/generated/
-        args: ["--py38-plus"]
+        args: ["--py312-plus"]
```

```suggestion
        args: ["--py39-plus"]
```

We actually do want to keep this at py39-plus since we still support python 3.9.  This may result in you having to make some other fixes below.